core/src/main/scala/cats/data/Nested.scala

```diff
@@ -81,6 +81,16 @@ private[data] sealed abstract class NestedInstances3 extends NestedInstances4 {
 }
 
 private[data] sealed abstract class NestedInstances4 extends NestedInstances5 {
+  implicit def catsDataApplicativeErrorForNested[F[_]: ApplicativeError[?[_], E], G[_]: Applicative, E]: ApplicativeError[Nested[F, G, ?], E] =
```

Comment:
couldn't you do it both ways? Put the `F[_]: ApplicativeError` or the `G[_]: ApplicativeError`?

Like: `Nested[Validated[String, ?], Option, ?]` or `Nested[Option, Validate[String, ?], ?]` You could have put the error on the inside or out, no?

Reply:
could we make two implementations at different priorities?

Reply:
I see your comment about reversing being hard unless you have `F[_]: Monad`. Indeed.

I wonder if it is worth adding that instance (`F[_]: Monad` and `G[_]: ApplicativeError`).

Reply:
Generally, I don't think we often have a constellation like that, but yeah for something like `Nested[Option, Validated[String, ?], ?]` it would work. However, I don't think it's enough to implement it on `Nested`. In my comment, I used `F[Validated[E, A]]` (aka `ValidatedT[F, E, A]`), but that has the extra property of being able to fold the `Validated` inside.

If all we have is `ApplicativeError`, we can't get there, because we don't have a function from `E => G[A]`, we only have `E => F[G[A]]`

```scala
// We only have this
def handleErrorWith(ga: G[A])(f: E => G[A]): G[A]

// but we need to implement this:
def handleErrorWith(fa: Nested[F, G, A])(f: E => Nested[F, G, A]): Nested[F, G, A]
```
We can't use the `f: E => Nested[F, G, A]` for the inner `ApplicativeError`, since we have no way to turn a `Nested[F, G, A]` into a `G[A]`

 The only way to do that, would be to give F and G `Traverse` instances or F an extra `Comonad` instance. Both of which would make this whole thing not really worth it IMO.
 
We can do it with `Validated`, because it allows us to `fold` it, but that's not true for the generic `ApplicativeError`.

Of course, I might be completely wrong, so point me in the right direction if I made a wrong assumption somewhere. 😄